ballista/rust/scheduler/src/state/task_manager.rs

```diff
@@ -383,18 +383,37 @@ impl<T: 'static + AsLogicalPlan, U: 'static + AsExecutionPlan> TaskManager<T, U>
     }
 
     pub async fn executor_lost(&self, executor_id: &str) -> Result<()> {
+        let mut fail_jobs = vec![];
         {
             let job_cache = self.active_job_cache.read().await;
```

Comment:
`fail_job` will need to acquire a write lock so we need to collect all the jobs and drop the lock before calling it.